### PR TITLE
Make some es6 updates and fix eslint rules

### PR DESCRIPTION
**Pull request by weikinhuang**

### CODE REVIEW

File: src/uri.js [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
-    PCHAR = UNRESERVED + SUB_DELIMS + "\\:\\@",
-    SCHEME = ALPHA + DIGIT + "\\-\\+\\.",
-    AUTHORITY = PCHAR,
-    PATH = PCHAR + "\\/",
-    QUERY = PCHAR + "\\/\\?",
-    FRAGMENT = PCHAR + "\\/\\?",
-    URIREGEX = /^(?:([^:\/?#]+):)?(?:\/\/([^\/?#]*))?([^?#]*)(?:\?([^#]*))?(?:#(.*))?$/;
+const ALPHA = 'a-zA-Z';
+const DIGIT = '0-9';
+const GEN_DELIMS = "\\:\\/\\?\\#\\[\\]\\@"; // eslint-disable-line quotes
```
Comment (weikinhuang) [inline src/uri.js]: this is laziness on my part, will update in the following PR

File: src/client.js [hunk 2]
```diff
@@ -53,54 +41,52 @@ export default class Client {
...
+    if (this._descriptorCache[uri]) {
       return this._descriptorCache[uri];
     }
 
-    this._descriptorCache[uri] = xhr(uri, extend({}, this._options.walkOptions, { method: 'options' }))
-    .then(function(res) {
-      if (res.status < 200 || res.status >= 300) {
+    this._descriptorCache[uri] = xhr(uri, merge({}, this._options.walkOptions, { method: 'options' }))
+    .then((res) => {
+      if (!res.ok) {
```
Comment (weikinhuang) [inline src/client.js]: this is the same condition as `res.status < 200 || res.status >= 300`

File: src/resource.js [hunk 3]
```diff
@@ -58,13 +49,10 @@ export default class Resource {
...
-  _issueRequest(method, params, body, options) {
-    options = options || {};
-
+  _issueRequest(method, params, body, options = {}) {
     var selfConn = this.getConnection({ name: 'self', data: params || {} });
-    var requestOptions = extend({}, this._requestOptions, options, { method: method });
-    requestOptions.headers = extend({}, this._requestOptions.headers || {}, options.headers || {});
-    if (body) {
+    var requestOptions = merge({}, this._requestOptions, options, { method: method });
+    if (body && !requestOptions.body) {
```
Comment (weikinhuang) [inline src/resource.js]: this is technically new functionality

**Comment by agaripian** [comment]
LGTM

File: src/resource.js (hunk 3 above)
Comment (kaicataldo) [inline src/resource.js]: This is a change in behavior, right? Before it would set body regardless of whether it was set before or not?

Comment (weikinhuang) [inline src/resource.js]: yes

Comment (weikinhuang) [inline src/resource.js]: I mean i can undo this change if you want

File: src/uri.js [hunk 4]
```diff
@@ -1,15 +1,15 @@
-const ALPHA = "a-zA-Z";
-const DIGIT = "0-9";
-const GEN_DELIMS = "\\:\\/\\?\\#\\[\\]\\@";
-const SUB_DELIMS = "\\!\\$\\&\\'\\(\\)\\*\\+\\,\\;\\=";
+const ALPHA = 'a-zA-Z';
+const DIGIT = '0-9';
+const GEN_DELIMS = "\\:\\/\\?\\#\\[\\]\\@"; // eslint-disable-line quotes
```
Comment (kaicataldo) [inline src/uri.js]: Why can't we change these?

Comment (weikinhuang) [inline src/uri.js]: laziness, I'll change it later

Comment (kaicataldo) [inline src/uri.js]: I see you answered above - sorry, was going commit by commit and it doesn't show comments

**Comment by kaicataldo** [comment]
:+1:

**Pull request merged**